catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/CachedSubscription.java

```diff
@@ -0,0 +1,184 @@
+
+    private static final String SUBSCRIPTION_ID_OSGI = "subscription-id";
+
+    private static final String EVENT_ENDPOINT = "event-endpoint";
+
+    private final SubscriptionMetadata metadata;
+
+    private transient Subscription subscription;
+
+    private transient ServiceRegistration registration;
+
+    public CachedSubscription(SubscriptionMetadata metadata) {
+        notNull(metadata, "Subscription metadata cannot be null");
+        this.metadata = metadata;
+
+        this.subscription = null;
+        this.registration = null;
+    }
+
+    public SubscriptionMetadata getMetadata() {
+        return metadata;
+    }
+
+    public Optional<Subscription> getSubscription() {
+        return Optional.ofNullable(subscription);
+    }
+
+    public boolean isType(String type) {
+        return type.equals(metadata.getTypeName());
+    }
+
+    public boolean isNotType(String type) {
+        return !isType(type);
+    }
+
+    public boolean isNotRegistered() {
+        return registration == null;
+    }
+
+    /**
+     * Attempt to construct a {@link Subscription} using the provided factory. If the resultant
+     * subscription is valid, register it as this cached subscription's OSGi service.
+     *
+     * @param factory subscription factory corresponding to this cached subscription's type.
+     * @throws SubscriptionRegistrationException if the factory can't operate on this cached subscription.
+     */
+    public void registerSubscription(SubscriptionFactory factory) {
+        if (isNotType(factory.getTypeName())) {
+            throw new SubscriptionRegistrationException(format(
+                    "Factory type mismatch for subscription type [%s]",
+                    metadata.getTypeName()));
+        }
+
+        LOGGER.debug("Regenerating subscription [ {} | {} | {} ]",
+                metadata.getId(),
+                metadata.getTypeName(),
+                metadata.getCallbackAddress());
+
+        Subscription subscription = factory.createSubscription(metadata);
+        if (subscription == null) {
+            throw new SubscriptionRegistrationException("Regenerated subscription cannot be null");
+        }
+        registerSubscription(subscription);
+    }
+
+    /**
+     * Register the provided {@link Subscription} as this cached subscription's OSGi service.
+     *
+     * @param subscription the subscription to register.
+     * @throws SubscriptionRegistrationException if registration failed.
+     */
+    public void registerSubscription(Subscription subscription) {
+        LOGGER.debug("Registering service [{}]", metadata.getId());
+
+        Dictionary<String, String> properties = new Hashtable<>();
+        properties.put(SUBSCRIPTION_ID_OSGI, metadata.getId());
+        properties.put(EVENT_ENDPOINT, metadata.getCallbackAddress());
+
+        registration = getBundleContext().registerService(Subscription.class.getName(),
```

Comment:
I would prefer consistency, and not to run around prefixing all state operations with `this.` since (1) IDEs are pretty good nowadays and (2) the less we have to read the better. I understand the ctor is an exceptional case. 

I renamed the argument in `registerSubscription(Subscription)` to "sub" so that I could remove the `this.` from the assignment in that method. Now we are consistent.